src/Polly.SharedSpecs/NoOp/NoOpAsyncSpecs.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Polly.Specs.Helpers;
+using Polly.Utilities;
+using Xunit;
+
+namespace Polly.Specs.NoOp
+{
+    public class NoOpAsyncSpecs
+    {
+        [Fact]
+        public void Should_not_throw_when_executing()
+        {
+            var policy = Policy.NoOpAsync();
+
+            policy.Awaiting(async p => await p.ExecuteAsync(() => Task.FromResult(0)))
+                .ShouldNotThrow();
+        }
+
```

Comment:
On the unit-test side, could be nice to have four-line(?) test expressing that `NoOpPolicy` does actually execute the delegate.  `bool executed = false;` flipped to `true` by executing deleg through policy

(similar elsewhere across async / TResult pattern)